core/src/main/java/org/elasticsearch/Build.java

```diff
@@ -43,8 +43,10 @@
         final String date;
         final boolean isSnapshot;
 
+        final String esPrefix = "elasticsearch-" + Version.CURRENT;
         final URL url = getElasticsearchCodebase();
-        if (url.toString().endsWith(".jar")) {
+        final String urlStr = url.toString();
+        if (urlStr.startsWith("file://") && (urlStr.endsWith(esPrefix + ".jar") || urlStr.endsWith(esPrefix + "-SNAPSHOT.jar"))) {
```

Comment:
Oh, I misread it (mobile phone, sorry). The only thing that needs to change then is file:// -> file:/.